ebill_paynet/models/account_invoice.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import logging
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = "account.move"
+
+    @api.onchange("transmit_method_id", "partner_id")
+    def _onchange_transmit_method_id(self):
+        if self.type not in ("out_invoice", "out_refund"):
+            return
+        paynet_method = self.env.ref("ebill_paynet.paynet_transmit_method")
+        if self.transmit_method_id == paynet_method:
+            # It is not called when selecting the customer ?
+            contract = self.get_active_contract(
+                self.partner_id, self.transmit_method_id
+            )
+            if contract:
+                self.invoice_partner_bank_id = (
+                    contract.paynet_service_id.partner_bank_id
+                )
+
+    def _export_invoice(self):
+        """Export invoice with the help of account_invoice_export module."""
+        paynet_method = self.env.ref("ebill_paynet.paynet_transmit_method")
+        if self.transmit_method_id != paynet_method:
+            return super()._export_invoice()
+        message = self.create_paynet_message()
+        if not message:
+            raise UserError(_("Error generating Paynet message"))
+        message.send_to_paynet()
+        self.invoice_exported = True
+        return "Paynet invoice generated and in state {}".format(message.state)
+
+    # TODO: Should go in base_ebill_payment_contract
+    @api.model
+    def get_active_contract(self, partner, transmit_method):
+        # FIXME: Search on non stored field
+        contract = self.env["ebill.payment.contract"].search(
+            [("is_valid", "=", True), ("partner_id", "=", partner.id)], limit=1,
+        )
+        if not contract:
+            _logger.error(
+                "Paynet contract for {} not found for {}".format(
+                    partner.name, transmit_method.name
+                )
+            )
+        return contract
+
+    def create_paynet_message(self):
```

Comment:
It has to be a private method as you are returning a recordset (not compatible with the JSON/XML-RPC layer):
```suggestion
    def _create_paynet_message(self):
```
(or return the ID of the message record but you'll have to rebrowse it after)